### ISSUE TITLE
"SQLite error 1: no such table", "Database was not used on the correct thread" when running tests in XCTest

### ISSUE THREAD

**Issue by rhult** [groue/GRDB.swift#1302]
### What did you do?

I am trying to set up unit tests for my app that uses GRDB and I'm running into two problems when doing this. Please let me know if you prefer two separate issues for each one (I originally planned to file an issue for the "no such table" issue, but ran into the threading one when trying to create a minimal test case).

### What did you expect to happen?

For the threading issue: After studying the documentation of GRDB, I expect to be able to do what I do in the tests here. I have no explicit threading going on, and even if XCTest uses threads somehow, there will only ever be one thread doing anything with the database as far as I can see.

For the "no such table" issue, I expect to be able to use the API from unit tests. I am suspecting there is something going on with regards to code living in different modules (app vs test) and that could affect type names that might be used for mapping table names (e.g. MyType vs MyApp.MyType)? But my debugging couldn't confirm this. My demo project tries to mix and match what is done in the app module vs the test code for this reason.

### What happened instead?

"Database was not used on the correct thread" and "SQLite error 1: no such table".

### Environment

GRDB flavor: GRDB
GRDB version: 6.5.0 (and 6.4.0)
Installation method: SPM
Xcode version: 14.1
Swift version: Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Platform(s) running GRDB: iOS simulator
macOS version running Xcode: 13.0.1

### Demo Project

https://github.com/rhult/reproduce-grdb-nosuchtable

Tests are here:
https://github.com/rhult/reproduce-grdb-nosuchtable/blob/main/NoSuchTableTests/NoSuchTableTests.swift

I tried to simplify as far as I could, please let me know if there is anything I can do to provide more information or help in any way.

**Comment by groue** [comment]
Hello @rhult,

"Database was not used on the correct thread" is a programmer error, so your app gets terminated. To help investigations, please attach the stack trace.

**Comment by rhult** [comment]
Hi @groue,

Thanks for the quick reply. I finally realized what was causing the strange issues - linking both the app and the test target with GRDB.

Both issues I saw (wrong thread and no such table) were caused by linking GRDB twice. Maybe this can help other users that run into this: the correct fix for a failing build of a unit test target ("No such module 'CSQLite'") is to simply link the test target with CSQLite (possible since 6.4.0 I believe).
